backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/workplace/WorkplaceUserAssignRepository.java

```diff
@@ -50,7 +50,27 @@ public Optional<WorkplaceId> getWorkplaceIdByUserId(@NonNull final UserId userId
 	public void create(@NonNull final WorkplaceAssignmentCreateRequest request)
 	{
 		final I_C_Workplace_User_Assign record = InterfaceWrapperHelper.newInstance(I_C_Workplace_User_Assign.class);
+		syncAndSave(request, record);
+	}
+
+	public void createOrUpdate(@NonNull final WorkplaceAssignmentCreateRequest request)
+	{
+		final Optional<WorkplaceId> workplaceId = getWorkplaceIdByUserId(request.getUserId());
+		final I_C_Workplace_User_Assign record;
+		if (workplaceId.isPresent())
+		{
+			record = InterfaceWrapperHelper.load(workplaceId.get(), I_C_Workplace_User_Assign.class);
```

Comment:
this looks weird to me....

Reply:
I think u should search by workspace id and user id!

Reply:
ofc, right. thx!